### PR TITLE
[driver] Disable -filelist option when invoking the swift-update tool.

### PR DESCRIPTION
**Pull request by akyrtzi**
This disables -filelist option from the driver when invoking the swift-update tool, which cannot handle it.

Resolves rdar://25320355

### CODE REVIEW

**Comment by akyrtzi** [comment]
@jrose-apple please review

File: test/Driver/driver-compile.swift [hunk 1]
```diff
@@ -113,6 +113,8 @@
 // FILELIST: -primary-file {{.*/(driver-compile.swift|empty.swift)}}
 // FILELIST: -output-filelist {{[^-]}}
 
+// UPDATE-CODE-NOT: -filelist
+// UPDATE-CODE-NOT: -output-filelist
```
Comment (jrose-apple) [inline test/Driver/driver-compile.swift]: These don't actually test that it doesn't appear anywhere in the output. You'll need to do a second FileCheck run for that.

**Comment by jrose-apple** [comment]
LGTM with the two changes mentioned.

**Comment by akyrtzi** [comment]
Good catch! See new commit.

**Pull request merged**